library/hls/src/main/java/com/google/android/exoplayer2/source/hls/playlist/HlsMasterPlaylist.java

```diff
@@ -94,6 +95,28 @@ public Variant(
       this.captionGroupId = captionGroupId;
     }
 
+    /**
+     * Construct a Variant with only an (optional) video Rendition (for example, EXT-X-I-FRAME-STREAM-INF
+     * only allows alternate VIDEO Renditions, these are suggested if the non-Iframe Variant includes
+     * alternate video Rendition but not required)
+     *
+     * @param url See {@link #url}.
+     * @param format See {@link #format}.
+     * @param videoGroupId See {@link #videoGroupId}.
+     */
+    public Variant(
+        Uri url,
+        Format format,
+        @Nullable String videoGroupId) {
+      this(
+          url,
+          format,
+          videoGroupId,
+          /* audioGroupId */null,
+          /* subtitleGroupId */null,
+          /* captionGroupId */null);
+    }
+
     /**
```

Comment:
Alternate 'VIDEO' renditions should be a TODO, it is in the spec but I've never seen it in the wild.